worker/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/google/uuid"
+	"github.com/minio/minio-go/v7"
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+const (
+	workerCount int = 2 // Avoid OOMKilled when processing too many images at the same time
+
+	inputBucketName  string = "uploaded"
+	outputBucketName string = "processed"
+	minioEndpoint    string = "minio:9000"
+	accessKey        string = "miniouser"
+	secretKey        string = "miniopassword"
+
+	mqUser       string = "rabbitmquser"
+	mqPassword   string = "rabbitmqpassword"
+	mqHost       string = "rabbitmq"
+	exchangeName string = "events"
+	queueName    string = "uploaded"
+)
+
+type UploadMessage struct {
+	Id        int    `json:"id"`
+	ObjectUrl string `json:"url"`
+}
+
+func main() {
+	minioClient := initMinio()
+
+	mqConnection, mqChannel := initMQ()
+	defer mqChannel.Close()
+	defer mqConnection.Close()
+
+	messages := initConsumer(mqChannel)
+
+	jobs := make(chan amqp.Delivery)
+
+	for w := 1; w <= workerCount; w++ {
+		go worker(w, jobs, minioClient)
```

Comment:
Maybe we need that to be added to the requirements, but I think instance-based scaling makes things much easier in the k8s world: Let k8s take care of scaling horizontally!

This also doesn't play nicely with the CPU resource limit of `1` in the deployment.

Reply:
I agree that horizontal scaling would make the code easier, but I also think that applications should be able to process things in parallel when needed.
 
As for the CPU limit, my understanding is that a limit of 1 does not imply only having only one core for the container. If I'm not mistaken there can still be work done across multiple cores even with a limit of 1, but the total CPU time cannot exceed 1 (i.e. there can be two cores doing 0.5 CPU's worth of work at the same time). But I'm not entirely sure about this!

Reply:
Your understanding of how the CPU limit works is correct. If you have a limit of 1 CPU, but 4 threads in your application, the 4 threads can run in parallel. However limits are implemented by time-slicing, so 4 threads exceed the 1 CPU limit in 0.25s and will then be preempted by the CPU scheduler and wait for 0.75s. 

This is especially fun when you operate interactive systems, because your latency will shoot up in unexpected ways when the system is under load (learned that the hard way).

Reply:
Interesting, I will have to think about that.